pkgs/racket-doc/syntax/scribblings/format.scrbl

```diff
@@ -0,0 +1,106 @@
+#lang scribble/manual
+
+@(require scribble/example
+          "common.rkt"
+          (for-label racket/function
+                     racket/format
+                     syntax/format
+                     syntax/srcloc))
+
+@(define make-format-eval (make-eval-factory '[syntax/format]))
+
+@title[#:tag "format"]{Creating Formatted Identifiers}
+@defmodule[syntax/format]
+
+@defproc[(~id [v (or/c identifier? string? symbol? keyword? char? number?)] ...
+              [#:context context (or/c syntax? #f) #f]
+              [#:source source source-location? context]
+              [#:props props (or/c syntax? #f 'infer) 'infer]
+              [#:track track (or/c #t (-> identifier? identifier?) #f) #t]
+              [#:binder? binder? any/c #t])
+         identifier?]{
+
+Builds an @tech[#:doc refman]{identifier} with a name constructed from the @racket[v]s, using
+@racket[context] for the @tech[#:doc refman]{lexical information} and @racket[source] for the source
+location. Each @racket[v] is converted to a symbol made up of the same characters that would be
+produced by @racket[~a], except for @tech[#:doc refman]{identifiers}, which use their symbolic name,
+and @tech[#:doc refman]{keywords}, which are converted using @racket[keyword->string] (i.e. the
+preceeding @litchar{#:} characters are not included).
+
+@(examples
+  #:eval (make-format-eval) #:once
+  (~id #'hyphenated "-" #'id)
+  (~id #'made #\- 'from #\- '#:many #\- "pieces"))
+
+If @racket[props] is a @tech[#:doc refman]{syntax object} or @racket[#f], then
+@tech[#:doc refman #:key "syntax property"]{syntax properties} from @racket[props] are copied to the
+new identifier in the same way as for @racket[datum->syntax]. If @racket[props] is @racket['infer],
```

Comment:
Should `props` be `prop`, to match the argument name in the docs for `datum->syntax`?

Reply:
I think `#:props` is a better name, and it’s what `format-id` already uses, so I’d prefer to keep it. I don’t think there will be any confusion due to the missing “s”, but I could clarify the wording somewhat if you feel it’s important.